### ISSUE TITLE
bug: Stencil occasionally does not generate dist/types directory

### ISSUE THREAD

**Issue by niklas2810** [ionic-team/stencil#5091]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.7.2

### Current Behavior

When some special conditions apply, Stencil sometimes fails to create the `dist/types` directory. This does **not** lead to a build error, which means that a broken version is generated if types are required in a downstream project.

Note: The missing types directory is the only noticable change, all other files looked the same.

### Expected Behavior

The types directory should always be generated.

### System Info

```shell
System: node 20.9.0
    Platform: windows (10.0.19045)
   CPU Model: Intel(R) Core(TM) i7-9850H CPU @ 2.60GHz (12 cpus)
    Compiler: O:\ws\temp\stencil-type-issue\node_modules\@stencil\core\compiler\stencil.js
       Build: 1699889895
     Stencil: 4.7.2
  TypeScript: 5.2.2
      Rollup: 2.42.3
      Parse5: 7.1.2
      Sizzle: 2.42.3
      Terser: 5.24.0

This has also been tested in different Linux environments (Ubuntu+Alpine).
```

### Steps to Reproduce

- Install `@stencil/sass` and use it as a plugin in your `stencil.config.ts`
- Make sure to use a `.scss` file for your component
- Add a property and mark it as internal using this comment above: the property `/** @internal */`
- Build the project multiple times and check whether the `types` directory is present in the `dist` folder (make sure to delete it before re-building, as Stencil does _not_ clear previous build results). For more convencience, a Python script has been added to the repository which performs the build and check automatically: `python loop.py 10` (runs the build 10 times and terminates if the types folder is missing).

### Code Reproduction URL

https://github.com/niklas2810/stencil-type-issue

### Additional Information

I know that this error seems rather cryptic, hopefully the reproduction repo can help with that. If you need more information, please let me know!

Best regards
Niklas

**Comment by rwaskiewicz** [comment]
Thanks! I was able to replicate this locally. I've labeled this to get it added to our internal backlog

**Comment by christian-bromann** [comment]
Released in [`v4.8.2`](https://github.com/ionic-team/stencil/releases/tag/v4.8.2)
